network/stream/intervals_test.go

```diff
@@ -298,38 +299,42 @@ func newTestExternalClient(netStore *storage.NetStore) *testExternalClient {
 	}
 }
 
-func (c *testExternalClient) NeedData(ctx context.Context, hash []byte) func(context.Context) error {
-	wait := c.netStore.FetchFunc(ctx, storage.Address(hash))
-	if wait == nil {
-		return nil
+func (c *testExternalClient) NeedData(ctx context.Context, key []byte) (bool, func(context.Context) error) {
+	fi, loaded, ok := c.netStore.GetOrCreateFetcher(ctx, key, "syncer")
+	if !ok {
+		return loaded, nil
 	}
+
 	select {
-	case c.hashes <- hash:
+	case c.hashes <- key:
 	case <-ctx.Done():
 		log.Warn("testExternalClient NeedData context", "err", ctx.Err())
-		return func(_ context.Context) error {
+		return false, func(_ context.Context) error {
 			return ctx.Err()
 		}
 	}
-	return wait
+
+	return loaded, func(ctx context.Context) error {
+		select {
```

Comment:
so should we not select on the `ctx.Done`?

Reply:
Why does this matter? Tests seem to be passing with the current `select`?